indispensable/src/commonMain/kotlin/at/asitplus/signum/indispensable/asn1/ObjectIdentifier.kt

```diff
@@ -58,40 +59,39 @@ class ObjectIdentifier @Throws(Asn1Exception::class) private constructor(
      * for details.
      * Lazily evaluated.
      */
-    val bytes: ByteArray by if (bytes != null) lazyOf(bytes) else lazy {
-        this.nodes.toOidBytes()
-    }
+    val bytes: ByteArray by lazy { bytes ?: this.nodes.toOidBytes() }
```

Comment:
i know. the old version was inefficient, because it initialized eagerly. why would we want this? are we missing early sanity checks if we init lazily?

Reply:
the class is initialized with either `bytes` or `nodes` set. why would we capture the constructor parameter in a lambda, then wait for the first access to actually return it, if we already know at construction time what the value will be?

Reply:
because maybe we will never access the bytes, but just operate on the nodes, if we require it for some intermediate FOO without actually encoding it

Reply:
but when you operate on the nodes, you lazy-init the nodes by looking at `this.bytes` (which lazy-inits `this.bytes` to the predetermined non-null ctor parameter)

Reply:
so the idea is that you always directly store whichever parameter is non-null, and the other parameter is lazy-initialized based on the non-null parameter

Reply:
this is still unaddressed

Reply:
then maybe i misunderstood. what is there to address?

Reply:
the change should be reverted; the field should be `lazy` initialized only if the constructor parameter is non-null